portal/common/lib/resource.bench.ts

```diff
@@ -1,21 +1,74 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { bench, describe, expect } from 'vitest';
+import { bench, describe, expect, vi, beforeEach } from 'vitest';
 import { fetchResource } from './resource';
-import { SuiClient, getFullnodeUrl } from '@mysten/sui/client'
-import { NETWORK } from './constants';
+import { SuiClient, SuiObjectData } from '@mysten/sui/client';
 import { isVersionedResource } from './types';
 import { SITES_USED_FOR_BENCHING } from './constants';
 
-const rpcUrl = getFullnodeUrl(NETWORK);
-const client = new SuiClient({ url: rpcUrl });
-describe('Resource fetching', () => {
+// Mock SuiClient methods.
+const getDynamicFieldObject = vi.fn();
+const getObject = vi.fn();
+
+const mockClient = {
+    getDynamicFieldObject,
+    getObject,
+} as unknown as SuiClient;
+
+vi.mock('./bcs_data_parsing', () => ({
+    ResourcePathStruct: vi.fn(),
+    ResourceStruct: vi.fn(),
+    DynamicFieldStruct: vi.fn(() => ({
+        parse: vi.fn().mockReturnValue({
+            value: {
+                blob_id: '0xresourceBlobId',
+                path: '/index.html',
+                content_type: 'text/html',
+                content_encoding: 'utf8',
+                blob_hash: 'mockedBlobHash',
+            },
+        }),
+    })),
+}));
+
+describe('Resource fetching with mocked network calls', () => {
+    beforeEach(() => {
+        vi.restoreAllMocks();
+
+        // Mock `getDynamicFieldObject` to provide data.
+        getDynamicFieldObject.mockResolvedValue({
+            data: {
+                objectId: '0xObjectId',
+                version: '1',
+                digest: 'mocked-digest',
+            },
+        });
+
+        // Mock `getObject` to provide BCS data.
+        getObject.mockResolvedValue({
+            data: {
+                objectId: '0xObjectId',
+                version: '1',
+                digest: 'mocked-digest',
+                bcs: {
+                    dataType: 'moveObject',
+                    bcsBytes: Buffer.from('valid-mocked-bcs-data', 'utf8').toString('base64'),
+                },
+            } as SuiObjectData,
+        });
+    });
+
     SITES_USED_FOR_BENCHING.forEach(([objectId, siteName]) => {
-        bench(`fetchResource: fetch the ${siteName} walrus site`, async () => {
+        // Benchmark the performance of fetching resources.
+        bench(`fetchResource: fetch the ${siteName} site`, async () => {
             const resourcePath = '/index.html';
-            const resp = await fetchResource(client, objectId, resourcePath, new Set());
+
+            // Use the mocked client and call fetchResource.
+            const resp = await fetchResource(mockClient, objectId, resourcePath, new Set());
+
+            // Validate that the fetched resource is versioned.
             expect(isVersionedResource(resp)).toBeTruthy();
         });
-    })
```

Comment:
`SITES_USED_FOR_BENCHING`: This will no longer be needed. We used it when we made real network requests to the actual sites. 

Now there should just be:
1. A test for an occasion like the landing page (with no redirects).
2. A test for a page **with** redirects (such as accessing a Flatlander).